gcloud-java-compute/src/main/java/com/google/cloud/compute/Operation.java

```diff
@@ -0,0 +1,846 @@
+    CheckingPeriod checkingPeriod = CheckingPeriod.getOrDefault(waitOptions);
+    long timeoutMillis = timeout.timeoutMillis();
+    Clock clock = options.clock();
+    long startTime = clock.millis();
+    while (!isDone()) {
+      if (timeoutMillis  != -1 && (clock.millis() - startTime)  >= timeoutMillis) {
+        throw new TimeoutException();
+      }
+      checkingPeriod.sleep();
+    }
+    return reload();
+  }
+
+  /**
+   * Fetches current operation's latest information. Returns {@code null} if the operation does not
+   * exist.
+   *
+   * @param options operation options
+   * @return an {@code Operation} object with latest information or {@code null} if not found
+   * @throws ComputeException upon failure
+   */
+  public Operation reload(OperationOption... options) {
+    return compute.getOperation(operationId, options);
+  }
+
+  /**
+   * Deletes this operation. Delete is only possible for operations that have completed their
+   * execution. Any attempt to delete a running operation will fail.
+   *
+   * @return {@code true} if operation was deleted, {@code false} if it was not found
+   * @throws ComputeException upon failure
+   */
+  public boolean delete() {
+    return compute.deleteOperation(operationId);
+  }
+
+  @Override
+  public String toString() {
+    return MoreObjects.toStringHelper(this)
+        .add("generatedId", generatedId)
+        .add("operationsId", operationId)
+        .add("clientOperationId", clientOperationId)
+        .add("operationType", operationType)
+        .add("targetLink", targetLink)
+        .add("targetId", targetId)
+        .add("status", status)
+        .add("statusMessage", statusMessage)
+        .add("user", user)
+        .add("progress", progress)
+        .add("insertTime", insertTime)
+        .add("startTime", startTime)
+        .add("endTime", endTime)
+        .add("errors", errors)
+        .add("warnings", warnings)
+        .add("httpErrorStatusCode", httpErrorStatusCode)
+        .add("httpErrorMessage", httpErrorMessage)
+        .add("description", description)
+        .toString();
+  }
+
+  @Override
+  public final int hashCode() {
+    return Objects.hash(operationId);
+  }
+
+  @Override
+  public final boolean equals(Object obj) {
+    if (obj == this) {
+      return true;
+    }
+    if (obj == null || !obj.getClass().equals(Operation.class)) {
+      return false;
+    }
+    Operation other = (Operation) obj;
+    return Objects.equals(toPb(), other.toPb()) && Objects.equals(options, other.options);
+  }
+
+  com.google.api.services.compute.model.Operation toPb() {
+    com.google.api.services.compute.model.Operation operationPb =
+        new com.google.api.services.compute.model.Operation();
+    if (generatedId != null) {
+      operationPb.setId(new BigInteger(generatedId));
+    }
+    operationPb.setName(operationId.operation());
+    operationPb.setClientOperationId(clientOperationId);
+    switch (operationId.type()) {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Switch statements should have a default label](https://www.codacy.com/app/mziccard/gcloud-java/file/2746909904/issues/source?bid=3123107&fileBranchId=3356751#l793)